query/common.d.ts

```diff
@@ -269,6 +277,20 @@ export interface Bindings extends Iterable<[RDF.Variable, RDF.Term]> {
   ) => Bindings;
 }
 
+export interface Paths extends Iterable<[RDF.Variable, RDF.Term][]> {
+  type: 'paths';
+  /**
+   * Check if a binding exist for the given variable.
+   * @param key A variable term or string. If it is a string, no `?` prefix must be given.
+   */
+  has: (key: number) => boolean;
+  /**
+   * Obtain the binding value for the given variable.
+   * @param key A variable term or string. If it is a string, no `?` prefix must be given.
+   */
+  get: (key: RDF.Variable | string) => RDF.Term | undefined;
```

Comment:
I think you can leave these methods to the individual Bindings objects that are within the Path. But you need to define a few other methods for a Path object, like:

```
[Symbol.iterator]: () => Iterator<Bindings>; // this is required by the Iterable interface I think

length: () => int; // nice to have

forEach: (fn: (value: Bindings) => any) => void;

nodes: () => Bindings[];
```

maybe some others to extend the path with a new node?

Maybe also define a `PathFactory`?

Reply:
I can add the symbol iterator, nodes function, and a "size" property to keep track of length. I cannot add a for Each function because there are no variable-value pairs in the Paths object.

Reply:
Yes but forEach will be iterating over Bindings objects for a Path.

Reply:
Why keep the `has` and `get` methods?